// Copyright 2017 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";



message ProposeAddRoleMember {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeRemoveRoleMember {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeAddRoleOwner {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeRemoveRoleOwner {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeAddRoleAdmin {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeRemoveRoleAdmin {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeAddRoleTask {

    string proposal_id = 1;

    string role_id = 2;

    string task_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ConfirmAddRoleTask {

    string proposal_id = 1;

    string role_id = 2;

    string task_id = 3;

    string reason = 4;

}


message RejectAddRoleTask {

    string proposal_id = 1;

    string role_id = 2;

    string task_id = 3;

    string reason = 4;
}


message ProposeRemoveRoleTask {

    string proposal_id = 1;

    string role_id = 2;

    string task_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ConfirmRemoveRoleTask {
    string proposal_id = 1;

    string role_id = 2;

    string task_id = 3;

    string reason = 4;

}


message RejectRemoveRoleTask {

    string proposal_id = 1;

    string role_id = 2;

    string task_id = 3;

    string reason = 4;
}


message ConfirmAddRoleMember {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message ConfirmRemoveRoleMember {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message ConfirmAddRoleOwner {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message ConfirmRemoveRoleOwner {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message ConfirmAddRoleAdmin {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message ConfirmRemoveRoleAdmin {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectAddRoleMember {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectRemoveRoleMember {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectAddRoleOwner {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectRemoveRoleOwner {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectAddRoleAdmin {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectRemoveRoleAdmin {

    string proposal_id = 1;

    string role_id = 2;

    string user_id = 3;

    string reason = 4;

}


message CreateRole {

    string role_id = 1;

    string name = 2;

    repeated string owners = 3;

    repeated string admins = 4;

    string metadata = 5;

}


message UpdateRole {

    string role_id = 1;

    string new_name = 2;

    string old_metadata_sha512 = 3;

    string new_metadata = 4;

}
